python/whylogs/core/resolvers.py

```diff
@@ -11,11 +12,45 @@
 ColumnSchema: TypeAlias = "ColumnSchema"  # type: ignore
 
 
+AdditionalMetric = namedtuple("AdditionalMetric", "metric column_names why_types")
+
+
 class Resolver(ABC):
     """A resolver maps from a column name and a data type to trackers.
 
     Note that the key of the result dictionaries defines the namespaces of the metrics in the serialized form."""
 
+    def __init__(self):
+        self.additional_metrics: List[AdditionalMetric] = []
+
+    def add_standard_metric(
+        self,
+        metric: StandardMetric,
+        column_names: List[String] = [],
+        why_types: List[Union[Fractional, Integral, String]] = [],
+    ):
+        if not column_names and not why_types:
+            raise ValueError("Either column names or why types must not be empty.")
+        if column_names and why_types:
+            raise ValueError("column_names or why_types should be defined, not both.")
```

Comment:
why can't I add both why_type and column_names? Maybe we don't need that flexibility but maybe its interesting to "Add `frequent_items` to columns named `dti_ratio` or columns of type `string`"

Reply:
yes, now passing both just raises a warning, saying that the name takes precedence.

Reply:
I thought about passing both when designing the `ResolverSpec`. One option would be to signal a warning or error if the specified column name didn't match the specified type (name AND type match semantics). OR match semantics seemed weird, but you can implement it with 2 `ResolverSpec`s if that's what you really want.